lib/streamlit/server/server.py

```diff
@@ -676,12 +680,31 @@ def _create_app_session(self, ws: WebSocketHandler) -> AppSession:
         """
         session_data = SessionData(self._main_script_path, self._command_line)
         local_sources_watcher = LocalSourcesWatcher(session_data)
+
+        is_public_cloud_app = False
+
+        try:
+            header_content = ws.request.headers["X-Streamlit-User"]
+            payload = base64.b64decode(header_content)
+            user_obj = json.loads(payload)
+            email = user_obj["email"]
+            is_public_cloud_app = user_obj["isPublicCloudApp"]
+        except (KeyError, binascii.Error, json.decoder.JSONDecodeError):
+            email = "test@localhost.com"
+
+        user_info: Dict[str, Optional[str]] = dict()
+        if is_public_cloud_app:
```

Comment:
Is this a data security-related check to prevent email collection for public apps? If so, this might not be sufficient since the developer can get access to the `email` information anyways (he can inject any code into the server to get this info). The only way to prevent this would be to not set the email in the `X-Streamlit-User` header on S4T side if it is a public app.

Reply:
Thanks @LukasMasuch 

We exclude email from the header in case of a private app, the code here is just handler code that event not access email field in case of public app